rust/src/dns/dns.rs

```diff
@@ -718,7 +734,7 @@ impl DNSState {
 
 const DNS_HEADER_SIZE: usize = 12;
 
-fn probe_header_validity(header: &DNSHeader, rlen: usize) -> (bool, bool, bool) {
+fn probe_header_validity(header: &DNSHeader, rlen: usize, dir: Direction) -> (bool, bool, bool) {
```

Comment:
I also think its time this function get some proper documentation as its no longer trivial.

Reply:
sure, but do you mean you agree with the changes ? (if there is nothing to change, should we document it ? )

Reply:
I agree with the changes, but as its got more complex I think it should now contain what all the constraints are as function documentation.